tools/ci_build/build.py

```diff
@@ -360,6 +360,10 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
                  "-Donnxruntime_DISABLE_CONTRIB_OPS=" + ("ON" if args.disable_contrib_ops else "OFF"),
                  "-Donnxruntime_MSVC_STATIC_RUNTIME=" + ("ON" if args.enable_msvc_static_runtime else "OFF"),
                  "-Donnxruntime_ENABLE_LANGUAGE_INTEROP_OPS=" + ("ON" if args.enable_language_interop_ops else "OFF"),
+                  # prometheus-cpp options that are included here due to CMP0077
+                  "-DENABLE_PUSH=OFF",
```

Comment:
This is risky. If any other component has the same macro defined and expect ON. This will trigger unexpected result.

Reply:
I agree, is there a better way to propagate options to Prometheus CPP? I could set and unset them in the relative areas but the due to CMP0077 I believe they have to be supplied in the CMAKE argument?

Reply:
Have you tried [this solution](https://stackoverflow.com/questions/3766740/overriding-a-default-option-value-in-cmake-from-a-parent-cmakelists-txt)?

Reply:
Looks to have worked, thank you!